Description: Fix LP: #1483410
 This patch corrects a simple bug in the test suite loader that was previously
 silently compesated by eariler versions of the standard library. The bug has
 caused us to effectively import modules and pacakges from _inside_ the
 checkbox_support package. For example, the module 'checkbox_support.dbus'
 was imported simply as 'dbus'. This caused all kinds of understandable havoc.
 The fix relies on the second, optional argument to defaultTestLoader.discover()
 which lets us specify that we want to load 'checkbox_support' from the parent
 directory.
Author: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>
Origin: upstream
Bug-Ubuntu: https://launchpad.net/bugs/1483410
Forwarded: no
Last-Update: 2015-08011

--- checkbox-support-0.20.orig/checkbox_support/tests/__init__.py
+++ checkbox-support-0.20/checkbox_support/tests/__init__.py
@@ -40,8 +40,9 @@ def load_unit_tests():
     """
     # Discover all unit tests. By simple convention those are kept in
     # python modules that start with the word 'test_' .
-    return defaultTestLoader.discover(
-        os.path.dirname(getabsfile(checkbox_support)))
+    start_dir = os.path.dirname(getabsfile(checkbox_support))
+    top_level_dir = os.path.normpath(os.path.join(start_dir, '..'))
+    return defaultTestLoader.discover(start_dir, top_level_dir=top_level_dir)
 
 
 def test_suite():
